index.d.ts

```diff
@@ -471,6 +471,48 @@ declare module '@amaas/amaas-core-sdk-js' {
     version: number
   }
 
+  // PNL (Profit & Loss)
+  export interface IPositionPNL {
+    assetId: string
+    assetManagerId: string
+    assetPnl: string
+    bookId: string
+    businessDate: Date
+    clientId: string
+    fxPnl: string
+    message: string
+    period: any
+    pnlStatus: string
+    pnlTimestamp: Date
+    quanity: number
+    realisedPnl: string
+    totalPnl: string
+    unrealisedPnl: string
+    createdBy: Date
+    updatedBy: Date
+    createdTime: Date
+    updatedTime: Date
+    version: string
+  }
```

Comment:
These are all typed as compulsory attributes. Some shouldn't be, like `createdBy`, `createdTime`, because you will want to instantiate the class without TypeScript forcing you to specify these attributes.

Some should be compulsory (like `assetManagerId`).

Also, you will need to add the API functions to this file, in the `api` namespace.

Reply:
Updated